File: Programmers/[68646] 풍선 터트리기/KingDonggyu/68646.js
function solution(a) {
  const n = a.length;
  let min = Infinity;

  for (let i = 0; i < n; i++) {
    min = Math.min(min, a[i]);
    a[i] = [a[i], min];
  }

  let answer = 0;
  min = Infinity;

  for (let i = n - 1; i > -1; i--) {
    min = Math.min(min, a[i][0]);
    if (a[i][0] <= a[i][1] || a[i][0] <= min) {
      answer++;
    }
  }

  return answer;
}

console.log(solution([-16, 27, 65, -2, 58, -92, -71, -68, -61, -33]));

File: Programmers/[68646] 풍선 터트리기/ONE/68646.java
import java.util.Arrays;

class Solution {
    public int solution(int[] a) {
        int n = a.length, answer = 2;
        int[] leftMin = Arrays.copyOf(a, n);
        int[] rightMin = Arrays.copyOf(a, n);

        for (int i = 1; i < n; i++) {
            leftMin[i] = Math.min(leftMin[i - 1], leftMin[i]);
            rightMin[n - 1 - i] = Math.min(rightMin[n - 1 - i], rightMin[n - i]);
        }

        for (int i = 1; i < n - 1; i++) {
            if (a[i] > leftMin[i - 1] && a[i] > rightMin[i + 1])
                continue;
            answer++;
        }

        return answer;
    }
}